test/K6/src/api/platform/authorization/maskinporten.js

```diff
@@ -0,0 +1,122 @@
+  var params = header.buildHeaderWithRuntimeAndJson(altinnToken, 'personal');
+  var body = [];
+  body.push(generatePolicyMatch(to, resourceid, orgnoOrPartyId));
+  var bodystring = JSON.stringify(body);
+  bodystring = bodystring.substring(1, bodystring.length-1)
+  return http.post(endpoint, bodystring, params);
+}
+
+/**
+ * POST call to revoke maskinportenschemas that have been received by the current party
+ * @param {*} altinnToken personal token for DAGL of receiver
+ * @param {*} coveredByPartyId the receiving party's partyid
+ * @param {*} from the offering organization's party id or organization number
+ * @param {*} resourceid the id of the resource to delegate
+ * @param {*} orgnoOrPartyId 'orgno' to set id to urn:altinn:organizationnumber
+ */
+export function revokeReceivedMaskinportenSchema(altinnToken, coveredByPartyId, from, resourceid, orgnoOrPartyId) {
+  var endpoint = config.buildMaskinPorteSchemaUrls(coveredByPartyId, 'revokereceived');
+  var params = header.buildHeaderWithRuntimeAndJson(altinnToken, 'personal');
+  var body = [];
+  body.push(generatePolicyMatch(from, resourceid, orgnoOrPartyId, true));
+  var bodystring = JSON.stringify(body);
+  bodystring = bodystring.substring(1, bodystring.length-1);
+  return http.post(endpoint, bodystring, params);
+}
+
+/**
+ * GET call to get maskinportenschemas that have been received by the current party
+ * @param {*} altinnToken personal token for DAGL
+ * @param {*} toPartyId party id or organization number of whom that offers the rule
+ * @param {*} to the receiving organization's party id or organization number
+ * @param {*} resourceid the id of the resource to delegate
+ * @param {*} orgnoOrPartyId 'orgno' to set id to urn:altinn:organizationnumber
+ */
+export function postMaskinportenSchema(altinnToken, offeredByPartyId, to, resourceid, orgnoOrPartyId) {
+  var endpoint = config.buildMaskinPorteSchemaUrls(offeredByPartyId, 'maskinportenschema');
+  var params = header.buildHeaderWithRuntimeAndJson(altinnToken, 'personal');
+  var body = [];
+  body.push(generatePolicyMatch(to, resourceid, orgnoOrPartyId));
+  var bodystring = JSON.stringify(body);
+  bodystring = bodystring.substring(1, bodystring.length-1)
+  return http.post(endpoint, bodystring, params);
+}
+
+/**
+ * function to build a policy match with action, user and resource details
+ * @param {*} toPartyId party id or organization number of whom that receives the rule
+ * @param {*} altinnAction read,write,sign
+ * @param {*} orgnoOrPartyId 'orgno' to set id to urn:altinn:organizationnumber
+ * @param {*} useFrom 'true', 'false'. If true, use 'from' in json, else use 'to'.
+ * @returns json object of a completed policy match
+ */
+function generatePolicyMatch(toPartyId, resourceid, orgnoOrPartyId, useFrom) {
+  var toId = 'urn:altinn:partyid';
+  if (orgnoOrPartyId == 'orgno') {
+    toId = 'urn:altinn:organizationnumber'
+  }
+  var toOrFrom = {
```

Comment:
toOrFrom? This sets up the to recipient as far as I see

Reply:
There's definitely a better way to do this, but if the parameter `useFrom` is true, the `toOrFrom` variable is pushed into `policyMatch.from`, otherwise it's pushed into `policyMatch.to`.